pyvo/mivot/features/sky_coord_builder.py

```diff
@@ -0,0 +1,177 @@
+    MangoRoles.PM_LONGITUDE: 'pm_ra_cosdec', MangoRoles.PM_LATITUDE: 'pm_dec',
+    MangoRoles.RADIAL_VELOCITY: 'radial_velocity', MangoRoles.EPOCH: 'obstime'}
+
+skycoord_param_galactic = {
+    MangoRoles.LONGITUDE: 'l', MangoRoles.LATITUDE: 'b', MangoRoles.PARALLAX: 'distance',
+    MangoRoles.PM_LONGITUDE: 'pm_l_cosb', MangoRoles.PM_LATITUDE: 'pm_b',
+    MangoRoles.RADIAL_VELOCITY: 'radial_velocity', MangoRoles.EPOCH: 'obstime'}
+
+
+class SkyCoordBuilder(object):
+    '''
+    Utility generating SkyCoord instances from MIVOT annotations
+
+    - SkyCoord instances can only be built from model classes containing the minimal
+    set of required parameters (a position).
+    - In this implementation, only the mango:EpochPosition class is supported since
+      it contains the information required to compute the epoch propagation which is a major use-case
+    '''
+
+    def __init__(self, mivot_instance_dict):
+        '''
+        Constructor
+
+        parameters
+        -----------
+        - mivot_instance_dict: viewer.MivotInstance.to_dict()
+                               Internal dictionary of the dynamic Python object generated
+                               from the MIVOT block
+        '''
+        self._mivot_instance_dict = mivot_instance_dict
+        self._map_coord_names = None
+
+    def build_sky_coord(self):
+        """
+        Build a SkyCoord instance from the dynamic Python object attribute.
+
+        returns
+        -------
+        - SkyCoord instance
+
+        raises
+        ------
+        - NoMatchingDMType: if the SkyCoord instance cannot be built.
+        """
+        if self._mivot_instance_dict and self._mivot_instance_dict["dmtype"] == "mango:EpochPosition":
+            return self._build_sky_coord_from_mango()
+        raise NoMatchingDMType("No INSTANCE with dmtype='mango:EpochPosition' has been found:"
+                               " cannot build a SkyCoord from annotations")
+
+    def _set_year_time_format(self, hk_field, besselian=False):
+        """
+        Format a date expressed in year as J-year
+
+        parameters
+        ----------
+        - hk_field: MIVOT instance parameter as a dict
+
+        returns
+        -------
+        - formatted string
+        """
+        scale = "J" if not besselian else "B"
+        return (f"{scale}{hk_field['value']}" if hk_field["unit"] in ("yr", "year")
+                else hk_field["value"])
+
+    def _get_space_frame(self, obstime=None):
+        """
+        Build an astropy space frame instance from the MIVOT annotations.
+
+        - Equinox are supported fot FK4/5
+        - Ref location are not supported
+
+        parameters
+        ----------
+        - obstime: string
+                   Observation time is given here because KF4 set an default value
+                   if it is not given
+        returns
+        -------
+        - Astropy space frame
```

Comment:
```suggestion
        Parameters
        ----------
        obstime : str
            Observation time is given here because KF4 set an default value
            if it is not given
                   
        Returns
        -------
        Astropy space frame
```

Reply:
done